File: app/src/main/java/com/example/jdgomes/desfrute/activity/MainActivity.java
package com.example.jdgomes.desfrute.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.widget.Toolbar;
import android.view.View;
import com.example.jdgomes.desfrute.R;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        carregaListaGastos();
        goToNewGastos();
    }

    protected void goToNewGastos() {
        FloatingActionButton fab = findViewById(R.id.btnAddGastos);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, GastosActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/jdgomes/desfrute/activity/MainActivity.kt
package com.example.jdgomes.desfrute.activity

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v7.widget.Toolbar
import com.example.jdgomes.desfrute.R

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        carregaListaGastos()
        goToNewGastos()
    }

    fun goToNewGastos() {
        val fab = findViewById<FloatingActionButton>(R.id.btnAddGastos)
        fab.setOnClickListener {
            val intent = Intent(this, GastosActivity::class.java)
            startActivity(intent)
        }
    }
}
